azure-pipelines.yml

```diff
@@ -0,0 +1,48 @@
+trigger:
+  branches:
+    include:
+    - master
+  tags:
+    include: ['*']
+
+strategy:
+  matrix:
+    linux:
+      imageName: 'ubuntu-16.04'
+    mac:
+      imageName: 'macos-10.13'
+    windows:
+      imageName: 'vs2017-win2016'
+
+pool:
+  vmImage: $(imageName)
+
+steps:
+
+- task: NodeTool@0
+  inputs:
+    versionSpec: '8.x'
+  displayName: 'Install Node.js'
+
+- bash: |
+    /usr/bin/Xvfb :99 -screen 0 1024x768x24 > /dev/null 2>&1 &
+    echo ">>> Started xvfb"
+  displayName: Start xvfb
+  condition: and(succeeded(), eq(variables['Agent.OS'], 'Linux'))
+
+- bash: |
+    echo ">>> Compile vscode-test"
+    yarn && yarn compile
+    echo ">>> Compiled vscode-test"
+    cd sample
+    echo ">>> Run sample integration test"
+    yarn && yarn compile && yarn test
+  displayName: Run Tests
+  env:
+    DISPLAY: ':99.0'
+
+- bash: |
```

Comment:
This is alright for `master`, but PR branches should just be tested. Anyway, we should probably publish the extension only when we'd like to, not with every commit. There must be a way to tell the pipeline when we'd like to actually go forward with the release and otherwise keep it as a draft. appVeyor/Travis allow this.

Reply:
The condition only allows this `deploy` _"target"_ to be executed for tags. Line 5 here:

```
01: condition: and(
02:     succeeded(),
03:     startsWith(
04:         variables['Build.SourceBranch'],
05:         'refs/tags/'
06:     ),
07:     eq(
08:         variables['Agent.OS'],
09:         'Linux'
10:     )
11: )
```